### PR TITLE
[ROCm] add case for FP32MatMulPattern skip property

### PR DESCRIPTION
**Pull request by jaglinux**
TF32 is not supported on ROCm and hence the torch/profiler/_pattern_matcher.py FP32MatMulPattern should return False for ROCm instead of checking the results of torch.cuda.get_arch_list().  Depending on the gfx arch running the test, test_profiler.py's test_profiler_fp32_matmul_pattern (__main__.TestExperimentalUtils) will fail otherwise.

Signed-off-by: Jagadish Krishnamoorthy <jagdish.krishna@gmail.com>

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport

### CODE REVIEW

**Comment by jaglinux** [comment]
cc @jeffdaily

**Review by jeffdaily — APPROVED** [review]

**Comment by jeffdaily** [comment]
@robieta please review, thanks.  CI failure is unrelated.

**Comment by jaglinux** [comment]
Corresponding tests are passing.
https://pipelines.actions.githubusercontent.com/serviceHosts/7d146c05-69c3-4c20-a0e7-818111670117/_apis/pipelines/1/runs/2469594/signedlogcontent/307?urlExpires=2022-12-12T20%3A25%3A32.6560498Z&urlSigningMethod=HMACV1&urlSignature=M%2BhYnJuPF%2BdX8LjKcDkVW51sX5Hx%2B7C28YJCPH8dxPM%3D

2022-12-12T08:59:31.9463247Z   test_profiler_fp32_matmul_pattern (__main__.TestExperimentalUtils) ... STAGE:2022-12-12 08:59:21 81595:81595 ActivityProfilerController.cpp:300] Completed Stage: Warm Up
2022-12-12T08:59:31.9464840Z STAGE:2022-12-12 08:59:23 81595:81595 ActivityProfilerController.cpp:306] Completed Stage: Collection
2022-12-12T08:59:31.9466459Z STAGE:2022-12-12 08:59:23 81595:81595 ActivityProfilerController.cpp:310] Completed Stage: Post Processing
2022-12-12T08:59:31.9467313Z ok (2.016s)


2022-12-12T08:59:31.9508175Z   test_profiler_pattern_matcher_json_report (__main__.TestExperimentalUtils) ... STAGE:2022-12-12 08:59:24 81595:81595 ActivityProfilerController.cpp:300] Completed Stage: Warm Up
2022-12-12T08:59:31.9509784Z STAGE:2022-12-12 08:59:24 81595:81595 ActivityProfilerController.cpp:306] Completed Stage: Collection
2022-12-12T08:59:31.9511213Z STAGE:2022-12-12 08:59:24 81595:81595 ActivityProfilerController.cpp:310] Completed Stage: Post Processing
2022-12-12T08:59:31.9512048Z ok (0.039s)

**Review by kit1980 — APPROVED** [review]

File: torch/profiler/_pattern_matcher.py [hunk 1]
```diff
@@ -300,9 +300,12 @@ def __init__(self, prof: profile, should_benchmark: bool = False):
 
     @property
     def skip(self):
-        # Anything less than sm_80 is not Ampere which doesn't support TF32
-        has_tf32 = all(
-            int(arch[3:]) >= 80 for arch in torch.cuda.get_arch_list())
+        if torch.version.hip:
```
Comment (kit1980) [inline torch/profiler/_pattern_matcher.py:303]: Nitpick: maybe `torch.version.hip is not None` for explicitness?

**Pull request closed without merging**